lib/astutils.cpp

```diff
@@ -273,11 +273,15 @@ bool isOppositeCond(bool isNot, bool cpp, const Token * const cond1, const Token
             if (isSameExpression(cpp, true, cond1->astOperand2(), cond2->astOperand2(), library, pure))
                 return isDifferentKnownValues(cond1->astOperand1(), cond2->astOperand1());
         }
-        if (Library::isContainerYield(cond1, Library::Container::EMPTY, "empty") && Library::isContainerYield(cond2->astOperand1(), Library::Container::SIZE, "size")) {
+        if (Library::isContainerYield(cond1, Library::Container::EMPTY, "empty") && 
+            Library::isContainerYield(cond2->astOperand1(), Library::Container::SIZE, "size") && 
+            cond1->astOperand1()->astOperand1()->varId() == cond2->astOperand1()->astOperand1()->astOperand1()->varId()) {
```

Comment:
sometimes I don't think there should be a variable id.

    void f(const std::string v[10]) {
        if (v[1].size() > 1) {
            if (v[2].empty()) {}
        }
    }

I would expect that `v` has a variable id but not `v[1]`.

I would expect that `Library::isContainerYield` should handle `v[1].empty()` but not sure if it does.. I don't get a FP for this code.

Reply:
So that case does fail. It will handle `v[1].empty()`, but it thinks `v[0]` and `v[1]` are the same variable.

Reply:
Sorry this is wrong. It doesn't wrongly diagnosis this case. However, it fails to diagnosis this case:

```cpp
void f1(const std::string v[10]) { if(v[0].size() > 42) if(v[0].empty()) {}}
```

It probably should check if the expressions are the same, but `isSameExpression` needs to stop at the `.`. I'll try to work on that, but I think this can be merged in now to fix the regression.

Reply:
For your information I have noticed that `ValueType` does not handle this yet. I plan to work on it:

    void f(std::vector<std::string> v) {
        if (v[0].empty()) {}
    }

In git head the `v[0]` has no type but it should have the type `std::string`.